### ISSUE TITLE
bug: Ion-datetime component is not accessible for the user using keyboard

### ISSUE THREAD

**Issue by kensodemann** [ionic-team/ionic-framework#21553]
# Bug Report

**Ionic version:**

[ ] **4.x**
[x] **5.x**

**Current behavior:**
The screen reads the ion-datetime component as a button. It’s not reading the associate aria-label or the <ion-label> for it. Also, the date selector is not accessible using only keyboard. It does not receive the focus by default.

Issue is reproducible from the sample implementation also
https://ionicframework.com/docs/demos/api/datetime/index.html?ionic:mode=ios

Refer to item 4 in the document shared in Slack for further details.
https://ionicteam.slack.com/archives/C7CGXAVTL/p1592319616186200

**Comment by kensodemann** [comment]
From #21602 - 

The tab navigation / arrow key navigation is not working for ion-datetime. This is an a11y issue.

This is reproducible with the example mentioned in ionic site: https://ionicframework.com/docs/demos/api/datetime/index.html?ionic:mode=ios

This is item 9 in the document that was shared in the internal Framework Slack channel

**Comment by kensodemann** [comment]
From the users we are working with on this:

the ion--datetime, have issues when we try on android device using talkback as well.
Following are the issues we could observe:

Navigation to the selection doesn't happen on swipe right.( next element navigation).

Navigation from month field to the date field and to the year doesn't happen on swipe right even after selection.

The label is not read out as provided.

**Comment by kalpeshjain89** [comment]
Hi,
We are using Ionic framework to develop a mobile app which needs to be accessible. The App will cater to countries having accessibility compliance and laws in place thereby making A11y a prime requirement. Our launch time is around mid May, 2021. We would appreciate if anyone can give a TAT for fixing this issue?

**Comment by willmartindev** [comment]
Thanks for the issue! `ion-datetime` has revamped keyboard navigation coming out in Framework v6.
